source/tutorials/packaging-existing-software.md

```diff
@@ -141,17 +173,20 @@ error:
 ```
 
 ### Finding the file hash
-As expected, the incorrect file hash caused an error, and Nix helpfully provided the correct one, which you can now substitute into `hello.nix` to replace `lib.fakeSha256`:
+As expected, the incorrect file hash caused an error, and Nix helpfully provided the correct one.
+Substitute substitute it into `hello.nix` to replace `lib.fakeSha256`:
```

Comment:
Nit: "Substitute X to replace Y" feels incorrect, since "substitute" and "replace" are the same action here. It would be simpler to write the following:

```suggestion
In `hello.nix`, replace `lib.fakeSha256` with the correct hash:
```